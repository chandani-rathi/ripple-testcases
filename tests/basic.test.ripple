import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { mount, flushSync, effect } from 'ripple';

describe('basic', () => {
	let container;

	function render(component) {
		mount(component, {
			target: container
		});
	}

	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		document.body.removeChild(container);
		container = null;
	});

	it('render static text', () => {
		component Basic() {
			<div>{'Hello World'}</div>
		}

		render(Basic);
		expect(container).toMatchSnapshot();
	});
});
