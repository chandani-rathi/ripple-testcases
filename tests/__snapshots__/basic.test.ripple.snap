// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic > render static text 1`] = `
<div>
  <div>
    Hello World
  </div>
  
</div>
`;
